### ISSUE TITLE
Error during event synchronization with Radicale server

### ISSUE THREAD

**Issue by ayevee** [aluxnimm/outlookcaldavsynchronizer#210]
When I try to synchronize event created in Outlook with Radicale, I get an error with the following raw report:

```
<?xml version="1.0" encoding="utf-16"?>
<SynchronizationReport xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <ProfileName>kouvykov-calendar</ProfileName>
  <ProfileId>8b058287-d930-4839-9726-629d06502312</ProfileId>
  <StartTime>2017-09-11T13:21:44.8386098Z</StartTime>
  <ADelta>Unchanged: 0 , Added: 1 , Deleted 0 ,  Changed 0</ADelta>
  <BDelta>Unchanged: 0 , Added: 0 , Deleted 0 ,  Changed 0</BDelta>
  <AJobsInfo>Create 0 , Update 0 , Delete 0</AJobsInfo>
  <BJobsInfo>Create 1 , Update 0 , Delete 0</BJobsInfo>
  <LoadErrors />
  <EntitySynchronizationReports>
    <EntitySynchronizationReport>
      <MappingErrors />
      <MappingWarnings />
      <AId>0000000060C8C30D1B327142B73B2CE9EA88B6FB070078F0B2343C6F754BA0A4D225F2F8726E0000000003430000AACCF31EF87A9E499047DED739ED86EC0000000014250000</AId>
      <ExceptionThatLeadToAbortion>CalDavSynchronizer.DataAccess.WebDavClientException: Response status code does not indicate success: '400' ('Bad Request'). Message:
Bad Request
   at CalDavSynchronizer.DataAccess.HttpClientBasedClient.WebDavClient.&lt;EnsureSuccessStatusCode&gt;d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CalDavSynchronizer.DataAccess.HttpClientBasedClient.WebDavClient.&lt;ExecuteWebDavRequest&gt;d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CalDavSynchronizer.DataAccess.HttpClientBasedClient.WebDavClient.&lt;ExecuteWebDavRequestAndReturnResponseHeaders&gt;d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CalDavSynchronizer.DataAccess.CalDavDataAccess.&lt;CreateNewEntity&gt;d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CalDavSynchronizer.Implementation.CalDavRepository`1.&lt;Create&gt;d__16.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GenSync.EntityRepositories.BatchEntityRepositoryAdapter`4.&lt;PerformOperations&gt;d__3.MoveNext()</ExceptionThatLeadToAbortion>
    </EntitySynchronizationReport>
  </EntitySynchronizationReports>
  <ConsiderExceptionThatLeadToAbortionAsWarning>false</ConsiderExceptionThatLeadToAbortionAsWarning>
  <Duration>00:00:00.5317070</Duration>
</SynchronizationReport>
```

As discussed in Kozea/Radicale#701 the issue is with DTSTART in VTIMEZONE being invalid:

> ```
> BEGIN:VTIMEZONE
> ...
> BEGIN:STANDARD
> DTSTART;VALUE=DATE:20110101
> ```
>
> The file is invalid. `VALUE=DATE` is not allowed here. `DTSTART` must be a date with local time value.
>
> [rfc5545-3.6.5](https://tools.ietf.org/html/rfc5545#section-3.6.5):
>
>       The mandatory "DTSTART" property gives the effective onset date
>      and local time for the time zone sub-component definition.
>      "DTSTART" in this usage MUST be specified as a date with a local
>      time value.

**Comment by aluxnimm** [comment]
Looks like a bug in our used iCalendar library.
But you can easily avoid this error when you change the timezone settings in event mapping configuration of the sync profile (advanced settings, open the tree view)
There you can enable "Create events on server with downloaded IANA timezones" and choose your timezone in the dropdown. Then you will have the official definitions from tzurl.org.

**Comment by ayevee** [comment]
@aluxnimm, thanks. The work-around with "Create events on server with downloaded IANA timezones" helped.

**Comment by aluxnimm** [comment]
Will try to fix it for the Windows timezones as well of course, what timezone did generate the VALUE=DATE? Moscow?

**Comment by ayevee** [comment]
> what timezone did generate the VALUE=DATE? Moscow?
Yeah, Europe/Moscow

Here is the full iCal output (from Radicale debug log):

```
2017-09-11 15:02:00,574 - [49184460] INFO: Successful login: 'ivan'
2017-09-11 15:02:00,578 - [49184460] DEBUG: Request content:
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//ddaysoftware.com//NONSGML DDay.iCal 1.0//EN
BEGIN:VTIMEZONE
TZID:Russian Standard Time
BEGIN:STANDARD
DTSTART:19701025T030000
RRULE:FREQ=YEARLY;UNTIL=20101231T210000Z;BYDAY=-1SU;BYHOUR=3;BYMINUTE=0;BYM
 ONTH=10
TZNAME:Russia TZ 2 Standard Time
TZOFFSETFROM:+0400
TZOFFSETTO:+0300
END:STANDARD
BEGIN:STANDARD
DTSTART;VALUE=DATE:20110101
RRULE:FREQ=YEARLY;UNTIL=20111231T210000Z;BYDAY=1SA;BYHOUR=0;BYMINUTE=0;BYMO
 NTH=1
TZNAME:Russia TZ 2 Standard Time
TZOFFSETFROM:+0400
TZOFFSETTO:+0300
END:STANDARD
BEGIN:STANDARD
DTSTART;VALUE=DATE:20120102
RRULE:FREQ=YEARLY;UNTIL=20121231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMO
 NTHDAY=1
TZNAME:Russia TZ 2 Standard Time
TZOFFSETFROM:+0300
TZOFFSETTO:+0300
END:STANDARD
BEGIN:STANDARD
DTSTART;VALUE=DATE:20130102
RRULE:FREQ=YEARLY;UNTIL=20131231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMO
 NTHDAY=1
TZNAME:Russia TZ 2 Standard Time
TZOFFSETFROM:+0300
TZOFFSETTO:+0300
END:STANDARD
BEGIN:STANDARD
DTSTART:20141026T020000
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYHOUR=2;BYMINUTE=0;BYMONTH=10
TZNAME:Russia TZ 2 Standard Time
TZOFFSETFROM:+0400
TZOFFSETTO:+0300
END:STANDARD
BEGIN:DAYLIGHT
DTSTART:19700329T020000
RRULE:FREQ=YEARLY;UNTIL=20101231T210000Z;BYDAY=-1SU;BYHOUR=2;BYMINUTE=0;BYM
 ONTH=3
TZNAME:Russia TZ 2 Summer Time
TZOFFSETFROM:+0300
TZOFFSETTO:+0400
END:DAYLIGHT
BEGIN:DAYLIGHT
DTSTART:20110327T020000
RRULE:FREQ=YEARLY;UNTIL=20111231T210000Z;BYDAY=-1SU;BYHOUR=2;BYMINUTE=0;BYM
 ONTH=3
TZNAME:Russia TZ 2 Summer Time
TZOFFSETFROM:+0300
TZOFFSETTO:+0400
END:DAYLIGHT
BEGIN:DAYLIGHT
DTSTART;VALUE=DATE:20120101
RRULE:FREQ=YEARLY;UNTIL=20121231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMO
 NTHDAY=1
TZNAME:Russia TZ 2 Summer Time
TZOFFSETFROM:+0300
TZOFFSETTO:+0300
END:DAYLIGHT
BEGIN:DAYLIGHT
DTSTART;VALUE=DATE:20130101
RRULE:FREQ=YEARLY;UNTIL=20131231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMO
 NTHDAY=1
TZNAME:Russia TZ 2 Summer Time
TZOFFSETFROM:+0300
TZOFFSETTO:+0300
END:DAYLIGHT
BEGIN:DAYLIGHT
DTSTART;VALUE=DATE:20140101
RRULE:FREQ=YEARLY;UNTIL=20141231T210000Z;BYDAY=1WE;BYHOUR=0;BYMINUTE=0;BYMO
 NTH=1
TZNAME:Russia TZ 2 Summer Time
TZOFFSETFROM:+0300
TZOFFSETTO:+0400
END:DAYLIGHT
END:VTIMEZONE
BEGIN:VEVENT
CLASS:PUBLIC
DESCRIPTION:Гарантийная книжка \nСТС \nСкидочная карта \n
DTEND;TZID=Russian Standard Time:20130330T160000
DTSTAMP:20170911T120201Z
DTSTART;TZID=Russian Standard Time:20130330T130000
LOCATION:Субару-центр Химки
PRIORITY:5
SEQUENCE:0
SUMMARY:ТО
TRANSP:TRANSPARENT
UID:43d71f30-42ff-49ac-b953-f59ddd73a4a1
X-MICROSOFT-CDO-BUSYSTATUS:FREE
END:VEVENT
END:VCALENDAR

2017-09-11 15:02:00,682 - [49184460] WARNING: Bad PUT request on '/ivan/7fafff2d-dc29-0723-d801-0dff332dd32c/43d71f30-42ff-49ac-b953-f59ddd73a4a1.ics': "In transformToNative, unhandled exception on line None: <class 'ValueError'>: unsupported DTSTART parm: VALUE=DATE (<VTIMEZONE| [<TZID{}Russian Standard Time>, <STANDARD| [<DTSTART{}19701025T030000>, <RRULE{}FREQ=YEARLY;UNTIL=20101231T210000Z;BYDAY=-1SU;BYHOUR=3;BYMINUTE=0;BYMONTH=10>, <TZNAME{}Russia TZ 2 Standard Time>, <TZOFFSETFROM{}+0400>, <TZOFFSETTO{}+0300>]>, <STANDARD| [<DTSTART{'VALUE': ['DATE']}20110101>, <RRULE{}FREQ=YEARLY;UNTIL=20111231T210000Z;BYDAY=1SA;BYHOUR=0;BYMINUTE=0;BYMONTH=1>, <TZNAME{}Russia TZ 2 Standard Time>, <TZOFFSETFROM{}+0400>, <TZOFFSETTO{}+0300>]>, <STANDARD| [<DTSTART{'VALUE': ['DATE']}20120102>, <RRULE{}FREQ=YEARLY;UNTIL=20121231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMONTHDAY=1>, <TZNAME{}Russia TZ 2 Standard Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0300>]>, <STANDARD| [<DTSTART{'VALUE': ['DATE']}20130102>, <RRULE{}FREQ=YEARLY;UNTIL=20131231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMONTHDAY=1>, <TZNAME{}Russia TZ 2 Standard Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0300>]>, <STANDARD| [<DTSTART{}20141026T020000>, <RRULE{}FREQ=YEARLY;BYDAY=-1SU;BYHOUR=2;BYMINUTE=0;BYMONTH=10>, <TZNAME{}Russia TZ 2 Standard Time>, <TZOFFSETFROM{}+0400>, <TZOFFSETTO{}+0300>]>, <DAYLIGHT| [<DTSTART{}19700329T020000>, <RRULE{}FREQ=YEARLY;UNTIL=20101231T210000Z;BYDAY=-1SU;BYHOUR=2;BYMINUTE=0;BYMONTH=3>, <TZNAME{}Russia TZ 2 Summer Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0400>]>, <DAYLIGHT| [<DTSTART{}20110327T020000>, <RRULE{}FREQ=YEARLY;UNTIL=20111231T210000Z;BYDAY=-1SU;BYHOUR=2;BYMINUTE=0;BYMONTH=3>, <TZNAME{}Russia TZ 2 Summer Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0400>]>, <DAYLIGHT| [<DTSTART{'VALUE': ['DATE']}20120101>, <RRULE{}FREQ=YEARLY;UNTIL=20121231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMONTHDAY=1>, <TZNAME{}Russia TZ 2 Summer Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0300>]>, <DAYLIGHT| [<DTSTART{'VALUE': ['DATE']}20130101>, <RRULE{}FREQ=YEARLY;UNTIL=20131231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMONTHDAY=1>, <TZNAME{}Russia TZ 2 Summer Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0300>]>, <DAYLIGHT| [<DTSTART{'VALUE': ['DATE']}20140101>, <RRULE{}FREQ=YEARLY;UNTIL=20141231T210000Z;BYDAY=1WE;BYHOUR=0;BYMINUTE=0;BYMONTH=1>, <TZNAME{}Russia TZ 2 Summer Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0400>]>]>)"
Traceback (most recent call last):
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/vobject/base.py", line 185, in transformToNative
    return self.behavior.transformToNative(self)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/vobject/icalendar.py", line 1017, in transformToNative
    obj.registerTzinfo(obj.tzinfo)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/vobject/icalendar.py", line 155, in gettzinfo
    return tz.tzical(buffer).get()
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/dateutil/tz/tz.py", line 1154, in __init__
    self._parse_rfc(fobj.read())
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/dateutil/tz/tz.py", line 1275, in _parse_rfc
    cache=True)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/dateutil/rrule.py", line 1605, in __call__
    return self._parse_rfc(s, **kwargs)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/dateutil/rrule.py", line 1564, in _parse_rfc
    raise ValueError("unsupported DTSTART parm: "+parm)
ValueError: unsupported DTSTART parm: VALUE=DATE

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/radicale/__init__.py", line 853, in do_PUT
    items = tuple(vobject.readComponents(content or ""))
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/vobject/base.py", line 1125, in readComponents
    component.transformChildrenToNative()
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/vobject/base.py", line 673, in transformChildrenToNative
    child = child.transformToNative()
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/vobject/base.py", line 198, in transformToNative
    raise ParseError(msg, lineNumber)
vobject.base.ParseError: "In transformToNative, unhandled exception on line None: <class 'ValueError'>: unsupported DTSTART parm: VALUE=DATE (<VTIMEZONE| [<TZID{}Russian Standard Time>, <STANDARD| [<DTSTART{}19701025T030000>, <RRULE{}FREQ=YEARLY;UNTIL=20101231T210000Z;BYDAY=-1SU;BYHOUR=3;BYMINUTE=0;BYMONTH=10>, <TZNAME{}Russia TZ 2 Standard Time>, <TZOFFSETFROM{}+0400>, <TZOFFSETTO{}+0300>]>, <STANDARD| [<DTSTART{'VALUE': ['DATE']}20110101>, <RRULE{}FREQ=YEARLY;UNTIL=20111231T210000Z;BYDAY=1SA;BYHOUR=0;BYMINUTE=0;BYMONTH=1>, <TZNAME{}Russia TZ 2 Standard Time>, <TZOFFSETFROM{}+0400>, <TZOFFSETTO{}+0300>]>, <STANDARD| [<DTSTART{'VALUE': ['DATE']}20120102>, <RRULE{}FREQ=YEARLY;UNTIL=20121231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMONTHDAY=1>, <TZNAME{}Russia TZ 2 Standard Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0300>]>, <STANDARD| [<DTSTART{'VALUE': ['DATE']}20130102>, <RRULE{}FREQ=YEARLY;UNTIL=20131231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMONTHDAY=1>, <TZNAME{}Russia TZ 2 Standard Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0300>]>, <STANDARD| [<DTSTART{}20141026T020000>, <RRULE{}FREQ=YEARLY;BYDAY=-1SU;BYHOUR=2;BYMINUTE=0;BYMONTH=10>, <TZNAME{}Russia TZ 2 Standard Time>, <TZOFFSETFROM{}+0400>, <TZOFFSETTO{}+0300>]>, <DAYLIGHT| [<DTSTART{}19700329T020000>, <RRULE{}FREQ=YEARLY;UNTIL=20101231T210000Z;BYDAY=-1SU;BYHOUR=2;BYMINUTE=0;BYMONTH=3>, <TZNAME{}Russia TZ 2 Summer Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0400>]>, <DAYLIGHT| [<DTSTART{}20110327T020000>, <RRULE{}FREQ=YEARLY;UNTIL=20111231T210000Z;BYDAY=-1SU;BYHOUR=2;BYMINUTE=0;BYMONTH=3>, <TZNAME{}Russia TZ 2 Summer Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0400>]>, <DAYLIGHT| [<DTSTART{'VALUE': ['DATE']}20120101>, <RRULE{}FREQ=YEARLY;UNTIL=20121231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMONTHDAY=1>, <TZNAME{}Russia TZ 2 Summer Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0300>]>, <DAYLIGHT| [<DTSTART{'VALUE': ['DATE']}20130101>, <RRULE{}FREQ=YEARLY;UNTIL=20131231T210000Z;BYHOUR=0;BYMINUTE=0;BYMONTH=1;BYMONTHDAY=1>, <TZNAME{}Russia TZ 2 Summer Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0300>]>, <DAYLIGHT| [<DTSTART{'VALUE': ['DATE']}20140101>, <RRULE{}FREQ=YEARLY;UNTIL=20141231T210000Z;BYDAY=1WE;BYHOUR=0;BYMINUTE=0;BYMONTH=1>, <TZNAME{}Russia TZ 2 Summer Time>, <TZOFFSETFROM{}+0300>, <TZOFFSETTO{}+0400>]>]>)"
2017-09-11 15:02:00,819 - [49184460] DEBUG: Running hook
2017-09-11 15:02:00,855 - [49184460] DEBUG: Response content:
Bad Request
```

**Comment by aluxnimm** [comment]
Fixed in 1b2d5d2. Will be in next release.
